addons/l10n_in/tests/test_l10n_in_fiscal_position.py

```diff
@@ -132,3 +132,86 @@ def test_l10n_in_fiscal_in_bill_to_ship_to(self):
             out_invoice.fiscal_position_id,
             self.env['account.chart.template'].ref('fiscal_position_in_export_sez_in')
         )
+
+    def test_l10n_in_fiscal_in_vendor_bills(self):
+        self.env.company = self.default_company
+        company_res_state = self.default_company.state_id
+        template = self.env['account.chart.template']
+        self.partner_a.write({
+            'state_id': company_res_state.id,
+            'country_id': company_res_state.country_id.id,
+        })
+        other_state = self.env['res.country.state'].search([
+            ('id', '!=', company_res_state.id), ('country_id', '=', company_res_state.country_id.id)],
+            limit=1
+        )
+        # Intra State
+        vendor_bill_1 = self._assert_invoice_fiscal_position(
+            fiscal_position_ref='fiscal_position_in_intra_state',
+            partner=self.partner_a,
+            taxes=self.sgst_sale_18,
+            move_type='in_invoice',
+            post=False,
+        )
```

Comment:
Please at least separate the setup from the sub tests, and use some blank lines to visually help the reader
```suggestion
        other_state = self.env['res.country.state'].search([
            ('id', '!=', company_res_state.id), ('country_id', '=', company_res_state.country_id.id)],
            limit=1
        )
        
        # Intra State
        self.partner_a.write({
            'state_id': company_res_state.id,
            'country_id': company_res_state.country_id.id,
        })
        vendor_bill_1 = self._assert_invoice_fiscal_position(
            fiscal_position_ref='fiscal_position_in_intra_state',
            partner=self.partner_a,
            taxes=self.sgst_sale_18,
            move_type='in_invoice',
            post=False,
        )
```

Even better if you use `self.subTest` or even different tests completely since there is basically no setup